restservice/src/main/java/org/itevents/service/transactional/MyBatisUserService.java

```diff
@@ -137,4 +150,43 @@ public void checkPassword(User user, String password) {
             throw new WrongPasswordServiceException(message);
         }
     }
-}
+
+    @Override
+    public void setOtpToUser(User user, OtpGenerator otpGenerator) {
+        userDao.setOtpToUser(user, otpGenerator);
+    }
+
+    @Override
+    public User getUserByOtp(OtpGenerator otp) {
+        try {
+            return userDao.getUserByOtp(otp);
+        } catch (EntityNotFoundDaoException e) {
+            LOGGER.error(e.getMessage());
+            throw new EntityNotFoundServiceException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void sendEmailWithActivationLink(User user) throws Exception {
+        OtpGenerator otp = otpGenerator.generateOtp(1440);
```

Comment:
You have to exract this magic number into constant or (better) to create property and put it to `.properties` file